File: c++/codechef/3.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int t;
    cin>>t;
    while(t--){
        long long int n,k;
        cin>>n>>k;
        if(n<=k){
            cout<<"NO"<<endl;
            continue;
        }
        long long int sum = ((k)*(k+1))/2;
        if(sum<=n){
            cout<<"YES"<<endl;
        }
        else{
            cout<<"NO"<<endl;
        }
    }
}

File: c++/codechef/3.py
t = int(input())
while(t):
    t-=1
    n,k = map(int,input().split())
    if(n<=k):
        print("NO")
        continue
    s = (k*(k+1))/2
    if(s<=n):
        print("YES")
    else:
        print("NO")
